### PR TITLE
otpclient: init at 2.4.4

### PR DESCRIPTION
**Pull request by alexbakker**
###### Motivation for this change

New package for OTPClient, and two other new packages for libraries that it depends on.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by cole-h** [comment]
@ofborg eval

(Sorry, ofborg's certs recently had an issue, meaning that it was unable to
react to PRs. This has been fixed, but all PRs filed in this period will need
ofborg to be manually kicked off.)

File: pkgs/applications/misc/otpclient/default.nix [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "0zjvhcx9q8nsf97zikddxriky0kghi4j4i7312s94pl8c7kb4abr";
+  };
+
+  buildInputs = [ gtk3 jansson libgcrypt libzip libpng libcotp zbar ];
+  nativeBuildInputs = [ cmake pkg-config wrapGAppsHook ];
+
+  meta = with lib; {
+    description = "Highly secure and easy to use OTP client written in C/GTK that supports both TOTP and HOTP";
+    homepage = "https://github.com/paolostivanin/OTPClient";
+    license = licenses.gpl3;
```
Comment (yu-re-ka) [inline pkgs/applications/misc/otpclient/default.nix:33]: Please specify gpl3Only or gpl3Plus

**Review by yu-re-ka — COMMENTED** [review]
Otherwise looks good 👍🏻

**Comment by alexbakker** [comment]
Fixed. Thanks for taking a look!

File: pkgs/development/libraries/libcotp/default.nix [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    rev = "v${version}";
+    sha256 = "1qq4shwiz1if9vys052dnsbm4dfw1ynlj6nsb0v4zjly3ndspfsk";
+  };
+
+  buildInputs = [ libbaseencode libgcrypt ];
+  nativeBuildInputs = [ cmake pkg-config ];
+
+  meta = with lib; {
+    description = "C library that generates TOTP and HOTP";
+    homepage = "https://github.com/paolostivanin/libbaseencode";
```
Comment (yu-re-ka) [inline pkgs/development/libraries/libcotp/default.nix:19]: Forgot to change the homepage url when copying?

**Review by yu-re-ka — COMMENTED** [review]
Found one more nit, will merge when that is fixed

**Comment by alexbakker** [comment]
Whoops. Fixed.

**Pull request merged**